packages/algob/src/lib/tx.ts

```diff
@@ -96,208 +94,109 @@ export function encodeNote (note: string | undefined, noteb64: string| undefined
-
-  const signedTxn = txn.signTxn(from.sk);
-  const pendingTx = await deployer.algodClient.sendRawTransaction(signedTxn).do();
-  console.log(ALGO_MSG, {
-    from: from.addr,
-    to: receiver,
-    amount: amountMicroAlgos,
-    txid: pendingTx.txId
-  });
-  return await deployer.waitForConfirmation(pendingTx.txId);
 }
-/**
-  * Description:
-  * This function is used to transfer ASA
-  * from one account to another using fromAccount secret key (for signing)
-
-  * Returns:
-  * Transaction details
-*/
-
-export async function transferAsset (
-  deployer: AlgobDeployer,
-  assetId: number,
-  from: AccountSDK,
-  to: string,
-  amount: number
-): Promise<tx.ConfirmedTxInfo> {
-  const params = await deployer.algodClient.getTransactionParams().do();
-
-  const sender = from;
-  const recipient = to;
-  const revocationTarget = undefined;
-  const closeRemainderTo = undefined;
-  const note = undefined;
-
-  const xtxn = tx.makeAssetTransferTxnWithSuggestedParams(
-    sender.addr,
-    recipient,
-    closeRemainderTo,
-    revocationTarget,
-    amount,
-    note,
-    assetId,
-    params);
-
-  const rawSignedTxn = xtxn.signTxn(sender.sk);
-  const xtx = (await deployer.algodClient.sendRawTransaction(rawSignedTxn).do());
-  console.log("Transferring:", {
-    from: sender.addr,
-    to: recipient,
-    amount: amount,
-    assetID: assetId,
-    txId: xtx.txId
-  });
-  return await deployer.waitForConfirmation(xtx.txId);
-}
-
-/**
- * Description:
- * This function is used to transfer Algos
- * from one account to another using logic signature (for signing)
-
- * Returns:
- * Transaction details
-*/
 
-export async function transferMicroAlgosLsig (
-  deployer: AlgobDeployer,
-  fromAccount: Account,
-  toAccountAddr: string,
-  amountMicroAlgos: number,
-  lsig: LogicSig,
-  payFlags: TxParams): Promise<tx.ConfirmedTxInfo> {
-  const params = await mkSuggestedParams(deployer.algodClient, payFlags);
-
-  const receiver = toAccountAddr;
-  const note = encodeNote(payFlags.note, payFlags.noteb64);
-
-  const txn = tx.makePaymentTxnWithSuggestedParams(
-    fromAccount.addr, receiver, amountMicroAlgos, payFlags.closeRemainderTo, note, params);
-
-  const signedTxn = tx.signLogicSigTransactionObject(txn, lsig);
-  const txId = txn.txID().toString();
-  console.log(txId);
-  const pendingTx = await deployer.algodClient.sendRawTransaction(signedTxn.blob).do();
-  console.log(ALGO_MSG, {
-    from: fromAccount.addr,
-    to: receiver,
-    amount: amountMicroAlgos,
-    txid: pendingTx.txId
-  });
-  return await deployer.waitForConfirmation(pendingTx.txId);
+function signTransaction (txn: Transaction, txnParam: execParams): Uint8Array {
+  switch (txnParam.sign) {
+    case "sk": {
+      return txn.signTxn(txnParam.fromAccount.sk);
+    }
+    case "lsig": {
+      const logicsig = txnParam.lsig;
+      if (logicsig === undefined) {
+        throw new Error("Lsig undefined");
```

Comment:
Can we make this error a little more descriptive?

Reply:
can you please suggest something ?
how about `lsig for this transaction is not defined` ?

Reply:
`logic signature for this transaction was not passed or - is not defined`. How about this?

Reply:
yes